File: towers_of_hanoi.rb
#!/usr/bin/env ruby
# http://rosettacode.org/wiki/Towers_of_Hanoi

def m(n,f,t,v)
  return if n <= 0
  m(n-1,f,v,t) 
  p "Move from #{f} to #{t}"
  m(n-1,v,t,f)
end

m(4,1,2,3)    

File: towers_of_hanoi.swift
// http://rosettacode.org/wiki/Towers_of_Hanoi

func move(n:Int,from:Int,to:Int,via:Int) {
  if (n > 0) {
    move(n-1,from,via,to)
    println("Move from \(from) to \(to).")
    move(n-1,via,to,from)
  }
}

move(4,1,2,3)
